File: bot-api/dotnet/api/src/mapper/InitialPositionMapper.cs
namespace Robocode.TankRoyale.BotApi.Mapper;

static class InitialPositionMapper
{
    internal static Schema.InitialPosition Map(InitialPosition source)
    {
        if (source == null) return null;

        return new Schema.InitialPosition
        {
            X = source.X,
            Y = source.Y,
            Direction = source.Direction
        };
    }
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/mapper/InitialPositionMapper.kt
package dev.robocode.tankroyale.server.mapper

import dev.robocode.tankroyale.server.model.InitialPosition

object InitialPositionMapper {
    fun map(initialPosition: dev.robocode.tankroyale.schema.InitialPosition?): InitialPosition? {
        initialPosition?.apply {
            return InitialPosition(x, y, direction)
        }
        return null
    }
}
